Create(R_lang-Vectors): added documentation for R lang vectors

monieshravichandrran opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

Added the documentation of Vectors in R language
- The definition and description of vectors
- The way to access vectors
- The way to update vectors
- Concept of Vector Recycling

### Type of Change

Adding a new entry
Creation change: created a folder and file under R lang for vectors docs

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [ ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=1049) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=1049) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=1049) it.</sub>

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+  - 'paths/computer-science'
+---
+
+Vectors are used to store and use multiple data values of the same type that can be accessed later by subsequent code. In R, the indexing of the vectors will start from ‘1’ and not from ‘0’ unlike the other programming languages.
```
```suggestion
Vectors are used to store and use multiple data values of the same type that can be accessed later by subsequent code. In R, the indexing of vectors will start from ‘1’ and not from ‘0’ unlike other programming languages.
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+## Creating a Vector
+
+In R, vectors are created or initialised using the ‘c’ function which means combining different elements together.
+Vectors can be of different datatypes.
```
This should be rephrased to avoid confusion. As is, it may be read that a single vector can contain multiple datatypes.

SSwiniarski replied:
This still needs to be addressed.

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,108 @@
+numbers # output [1,2,3,4,5,6,7,8,9,10]
+```
+
+In R, Character vectors contain alphanumeric values and special characters.
````
```suggestion
In R, character vectors can contain alphanumeric values and special characters.
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,108 @@
+Apart from the 'c' function semicolon(:) can also be used to create a vector containing numbers in a range.
+
+```r
+# This vector numbers contains all the numbers between 1 and 10
````
```suggestion
# This vector contains all the numbers between 1 and 10
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,108 @@
+In R, Character vectors contain alphanumeric values and special characters.
+
+```r
+# This vector contains both alphabets and numerics and type of the vector would be character vector type
````
```suggestion
# This vector contains both alphabetic and numeric values, and type of the vector would be character vector type
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+
+## Accessing an element in a Vector
+
+There is need to access the elements in vector to perform operations on each elements. The most common way to access the vector elements is by using the Subscript operator [].
```
```suggestion
There is need to access the elements in vector to perform operations on each elements. The most common way to access the vector elements is by using the subscript operator [].
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+
+## Vector Recycling
+
+Vector Recycling is a process in which two vectors are involved in an operation, that operation needs the vectors to be of same length but the two vectors are of different length, in those cases, R repeats the elements of shorter vector to match the length of longer vector.
```
```suggestion
Vector Recycling is a process in which two vectors are involved in an operation where the operation needs the vectors to be of same length and the two vectors are of different length. In those cases, R repeats the elements of shorter vector to match the length of longer vector.
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+
+Vector Recycling is a process in which two vectors are involved in an operation, that operation needs the vectors to be of same length but the two vectors are of different length, in those cases, R repeats the elements of shorter vector to match the length of longer vector.
+
+Example for Vector Recycling using Addition Operation :
```
```suggestion
Example for Vector Recycling using addition:
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+
+Example for Vector Recycling using Addition Operation :
+
+Vector Addition requires the two vectors to be of same length.
```
```suggestion
Vector addition requires the two vectors to be of same length.
```

SSwiniarski reviewed (changes requested):
@monieshravichandrran, I did a first review and left a few comments.

monieshravichandrran commented:
Hi @SSwiniarski, I have fixed all the changes suggested in the first review.

Can you re-review my PR?

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,108 @@
+In R, character vectors can contain alphanumeric values and special characters.
+
+```r
+# This vector contains both alphabetic and numeric values, and type of the vector would be character vector type
````
```suggestion
# This vector contains both alphabetic and numeric values, and would be a character vector type
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+
+## Accessing an element in a Vector
+
+There is need to access the elements in vector to perform operations on each elements. The most common way to access the vector elements is by using the subscript operator [].
```
```suggestion
There is need to access the elements in vector to perform operations on each element. The most common way to access the vector elements is by using the subscript operator [].
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+
+## Vector Recycling
+
+Vector Recycling is a process in which two vectors are involved in an operation where the operation needs the vectors to be of same length and the two vectors are of different length. In those cases, R repeats the elements of shorter vector to match the length of longer vector.
```
```suggestion
Vector recycling is a process in which two vectors are involved in an operation where the operation needs the vectors to be of same length and the two vectors are of different length. In those cases, R repeats the elements of shorter vector to match the length of longer vector.
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+
+Vector Recycling is a process in which two vectors are involved in an operation where the operation needs the vectors to be of same length and the two vectors are of different length. In those cases, R repeats the elements of shorter vector to match the length of longer vector.
+
+Example for Vector Recycling using addition:
```
```suggestion
Example for vector recycling using addition:
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+numbers1 <- c(1,2,3,4,5,6)
+numbers2 <- c(3,4)
+sum <- numbers1 + numbers2 # output [4,6,6,8,8,10]
+# here 1 is added with 3 and then 2 is added with 4 now numbers2 vector end is reached
```
```suggestion
# Here 1 is added with 3 and then 2 is added with 4. Now numbers2 vector end is reached.
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+numbers2 <- c(3,4)
+sum <- numbers1 + numbers2 # output [4,6,6,8,8,10]
+# here 1 is added with 3 and then 2 is added with 4 now numbers2 vector end is reached
+# Now to complete the addition operation R repeats the smaller vector till the larger vector end is reached
```
```suggestion
# Now to complete the addition operation R repeats the smaller vector till the larger vector end is reached.
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+sum <- numbers1 + numbers2 # output [4,6,6,8,8,10]
+# here 1 is added with 3 and then 2 is added with 4 now numbers2 vector end is reached
+# Now to complete the addition operation R repeats the smaller vector till the larger vector end is reached
+# Therefore, 3 is added with 3 and 4 is added with 4 then 5 is added with 3 and 6 is added with 4
```
```suggestion
# Therefore, 3 is added with 3 and 4 is added with 4 then 5 is added with 3 and 6 is added with 4.
```

SSwiniarski commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,108 @@
+# here 1 is added with 3 and then 2 is added with 4 now numbers2 vector end is reached
+# Now to complete the addition operation R repeats the smaller vector till the larger vector end is reached
+# Therefore, 3 is added with 3 and 4 is added with 4 then 5 is added with 3 and 6 is added with 4
+# This process is called as Vector Recycling
```
```suggestion
# This process is called as vector recycling
```

SSwiniarski reviewed (changes requested):
@monieshravichandrran, I had a few more comments for you.

SSwiniarski reviewed (approved):
@monieshravichandrran So far so good, this is ready for a second review.

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,111 @@
+---
+Title: 'Vectors'
+Description: 'Vectors are used to store and use multiple data values of the same type.'
```
```suggestion
Description: 'Vectors are used to store multiple values of the same data type.'
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,111 @@
+
+## Creating a Vector
+
+In R, vectors are created or initialised using the ‘c’ function which means combining different elements of the same type together.
```
```suggestion
In R, vectors are made with the ‘c()’ function to combine different elements of the same type together.
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,111 @@
+marks <- c(90,87,100,79,91)
+```
+
+Apart from the 'c' function semicolon(:) can also be used to create a vector containing numbers in a range.
````
```suggestion
Apart from the 'c()' function, the semicolon (`:`) can be used to create a vector containing a range of numbers:
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,111 @@
+```r
+# This vector contains both alphabetic and numeric values, and would be a character vector type
+alphanumerics <- c('water','fire','ice',90)
+alphanumerics # output ['water','fire','ice','90']
````
```suggestion
alphanumerics # Output: ['water','fire','ice','90']
```

Dusch4593 reviewed (changes requested):
@monieshravichandrran I left you some comments/suggestions for my review but I will have more after you've implemented this batch of changes. 😄

monieshravichandrran commented:
@Dusch4593 Have fixed the changes suggested in 2nd review.

Can you pls re-review the PR?

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,119 @@
+Vectors can be of different datatypes.
+
+```r
+# This creates a vector of names (character vector)
+names <- c('Adam','John','Walter')
````
```suggestion
# Character vector
names <- c('Adam', 'John', 'Walter')

# Integer vector
marks <- c(90, 87, 100, 79, 91)

# Logical vector
booleans <- c(TRUE, FALSE, TRUE)
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,119 @@
+numbers[2] # Output: 49
+number[4] # Output: 36
+```
+
````
```suggestion

The output would look like this:

```

And then let's add a new \```shell snippet for the output, like so:

```shell
[1] 49
[1] 36
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,119 @@
+
+```r
+# Accessing vector elements using logical indexing.
+numbers <- c(1,2,3,4,5)
+numbers[numbers<2]
````
```suggestion
numbers <- c(1, 2, 3, 4, 5)
numbers[numbers < 4]
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,119 @@
+# Accessing vector elements using logical indexing.
+numbers <- c(1,2,3,4,5)
+numbers[numbers<2]
+# Output: 1
```
```suggestion
# Output: [1] 1 2 3
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,119 @@
+
+## Updating a Vector
+
+A vector element can be updated just by replacing the value using the subscript operator [] by specifying the index.
```
```suggestion
Vector elements can be updated by specifying the index with the subscript operator (`[]`) and replacing with the new value:
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,119 @@
+A vector element can be updated just by replacing the value using the subscript operator [] by specifying the index.
+
+```r
+numbers <- c(2,5,1,7,9)
+
+numbers[3] = 8
+numbers # Output: [2,5,8,7,9]
+numbers[1] = 0
+numbers # Output: [0,5,8,7,9]
````
```suggestion
numbers <- c(2, 5, 1, 7, 9)

cat("Original vector: ", numbers, "\n")

numbers[3] = 8
cat("Third element changed: ", numbers, "\n")

numbers[1] = 0
cat("First element changed: ", numbers)
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,111 @@
+numbers[1] = 0
+numbers # output [0,5,8,7,9] first element is updated to 0
+```
+
````
This suggestion still needs resolved. Also, I would add a new \```shell snippet below this text with the output:

```shell
Original vector:  2 5 1 7 9 
Third element changed:  2 5 8 7 9 
First element changed:  0 5 8 7 9
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,119 @@
+
+## Vector Recycling
+
+Vector recycling is a process in which two vectors are involved in an operation where the operation needs the vectors to be of same length and the two vectors are of different length. In those cases, R repeats the elements of shorter vector to match the length of longer vector.
```
```suggestion
Vector recycling is a process in which two vectors of different lengths are operated upon such that the elements of the shorter vector are repeated to match the length of the longer vector.
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,119 @@
+
+Vector recycling is a process in which two vectors are involved in an operation where the operation needs the vectors to be of same length and the two vectors are of different length. In those cases, R repeats the elements of shorter vector to match the length of longer vector.
+
+Example for vector recycling using addition:
```
```suggestion
### Example

The following example presents two cases of vector recycling using addition, which requires the two vectors to be equal in length.
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,119 @@
+
+Example for vector recycling using addition:
+
+Vector addition requires the two vectors to be of same length.
+
+Case 1: When the two vector lengths are equal.
```
```suggestion
Case 1: When the two vector lengths are equal:
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,119 @@
+Case 1: When the two vector lengths are equal.
+
+```r
+numbers1 <- c(1,2,3,4,5)
+numbers2 <- c(3,4,5,6,7)
+sum <- numbers1 + numbers2
+# Output: [4,6,8,10,12]
````
```suggestion
numbers_1 <- c(1, 2, 3, 4, 5)
numbers_2 <- c(3, 4, 5, 6, 7)
sum <- numbers_1 + numbers_2
sum
# Output: [1] 4 6 8 10 12
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,119 @@
+# Output: [4,6,8,10,12]
+```
+
+Case 2: When the vector lengths are not the same.
````
```suggestion
Case 2: When the vector lengths are unequal:
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,119 @@
+Case 2: When the vector lengths are not the same.
+
+```r
+numbers1 <- c(1,2,3,4,5,6)
+numbers2 <- c(3,4)
+sum <- numbers1 + numbers2 
+# Output [4,6,6,8,8,10]
````
```suggestion
numbers_1 <- c(1, 2, 3, 4, 5, 6)
numbers_2 <- c(3, 4)
sum <- numbers_1 + numbers_2
sum
# Output: [1] 4 6 6 8 8 10
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,119 @@
+sum <- numbers1 + numbers2 
+# Output [4,6,6,8,8,10]
+```
+Here 1 is added with 3 and then 2 is added with 4. Now numbers2 vector end is reached.
+
+Now to complete the addition operation R repeats the smaller vector till the larger vector end is reached.
+
+Therefore, 3 is added with 3 and 4 is added with 4 then 5 is added with 3 and 6 is added with 4.4
+This process is called as vector recycling.
````
```suggestion

Here, the first two elements of `numbers_1` are added to the first, and only, two elements of `numbers_2`. Then, to complete the addition operation, R repeats (or recycles) the elements of the smaller `numbers_2` vector (`3` and `4`) until the end of `numbers_1` is reached.
```

Let's always make sure to leave an empty line above and below code blocks and headers. 😄

Dusch4593 reviewed (changes requested):
@monieshravichandrran Finished with my review and left you some comments/suggestions. 😄

Dusch4593 commented:
Hey @monieshravichandrran just wanted to check in with you and see how everything is going with the PR? Anything we can do to help? The end of Hacktoberfest is coming up soon!

Dusch4593 commented:
@monieshravichandrran Did you have any questions or need any assistance with implementing the latest round of feedback? As a reminder, today is the last day to merge your PRs and have them count for Hacktoberfest. 👀

monieshravichandrran commented:
Hi @Dusch4593 am extremely sorry as I was not able caught up in other work

I have completed and addressed your review 

Can you please re review it

Thank you

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,141 @@
+numbers <- c(65, 49, 55, 36, 126)
+numbers[2]
+numbers[4]
+
```
Add three backticks ``` here.

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,141 @@
+
+## Accessing Vector Elements
+
+There is need to access the elements in vector to perform operations on each element. The most common way to access the vector elements is by using the subscript operator [].
```
```suggestion
The most common way to access vector elements is by using the subscript operator `[]`:
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,141 @@
+Vector recycling is a process in which two vectors of different lengths are operated upon such that the elements of the shorter vector are repeated to match the length of the longer vector.
+
+### Example
+The following example presents two cases of vector recycling using addition, which requires the two vectors to be equal in length.
```
```suggestion

The following example presents two cases of vector recycling using addition, which requires the two vectors to be equal in length.
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
```diff
@@ -0,0 +1,141 @@
+
+### Example
+The following example presents two cases of vector recycling using addition, which requires the two vectors to be equal in length.
+
+Vector addition requires the two vectors to be of same length.
+
+Case 1: When the two vector lengths are equal:
```
```suggestion

Case 1: When the two vector lengths are equal:
```

Dusch4593 commented on content/r/concepts/vectors/vectors.md:
````diff
@@ -0,0 +1,141 @@
+sum
+# Output: [1] 4 6 6 8 8 10
+```
+Here, the first two elements of `numbers_1` are added to the first, and only, two elements of `numbers_2`. Then, to complete the addition operation, R repeats (or recycles) the elements of the smaller `numbers_2` vector (`3` and `4`) until the end of `numbers_1` is reached.
````
```suggestion

Here, the first two elements of `numbers_1` are added to the first, and only, two elements of `numbers_2`. Then, to complete the addition operation, R repeats (or recycles) the elements of the smaller `numbers_2` vector (`3` and `4`) until the end of `numbers_1` is reached.
```

Dusch4593 reviewed (approved):
@monieshravichandrran Since it has been a few weeks since there was last activity on this PR, I went ahead and implemented the last batch of feedback in 1b87a8af688c2e3fefa49d365b815592a74f879a so that we may merge. 😄

Dusch4593 commented:
Congrats and thank you so much for this contribution to Codecademy Docs, @monieshravichandrran! 🚀 🎊

The entry should now be live: https://www.codecademy.com/resources/docs/r/vectors

Currently, your Codecademy profile is anonymously listed on this entry page. To make it visible, you'll need to ensure your GitHub username is added to your Codecademy profile, log out of Codecademy, and log back in through your GitHub account:

<img width="917" alt="Screen Shot 2022-11-15 at 9 36 16 AM" src="https://user-images.githubusercontent.com/23144845/201946656-5f6c595c-a766-4277-ac04-23c7557b5e56.png">

Looking forward to your next contribution to Docs. 😄

P.S. Look out in the["Issues" tab](https://github.com/Codecademy/docs/issues) in the coming weeks; we'd like to show more attention to our entries for the R language.
